src/java/neroxis/map/SCMap.java

```diff
@@ -376,6 +382,257 @@ public void addCubeMap(CubeMap cubeMap) {
+            float y;
+            if(heightmapBase.inBounds((int) x,  (int) z)) {
+                y = heightmapBase.getValueAt((int) x, (int) z);
+            } else {
+                y = 0; }
+            getDecal(i).setPosition(new Vector3f(x, y, z));
+            Vector3f scale = getDecal(i).getScale();
+            getDecal(i).setScale(new Vector3f(scale.x * contentScaler, scale.y, scale.z * contentScaler));
+        }
+        for (int i = 0; i < getHydroCount(); i++) {
+            float x = (float) (StrictMath.round((contentScaler * getHydro(i).getPosition().x + shiftX) + 0.5) - 0.5);
+            float z = (float) (StrictMath.round((contentScaler * getHydro(i).getPosition().z + shiftZ) + 0.5) - 0.5);
+            float y;
+            if(heightmapBase.inBounds((int) x,  (int) z)) {
+                y = heightmapBase.getValueAt((int) x, (int) z);
+            } else {
+                y = 0; }
+            getHydro(i).setPosition(new Vector3f(x, y, z));
+        }
+        for (int i = 0; i < getMexCount(); i++) {
+            float x = (float) (StrictMath.round((contentScaler * getMex(i).getPosition().x + shiftX) + 0.5) - 0.5);
+            float z = (float) (StrictMath.round((contentScaler * getMex(i).getPosition().z + shiftZ) + 0.5) - 0.5);
+            float y;
+            if(heightmapBase.inBounds((int) x,  (int) z)) {
+                y = heightmapBase.getValueAt((int) x, (int) z);
+            } else {
+                y = 0; }
+            getMex(i).setPosition(new Vector3f(x, y, z));
+        }
+        for (int i = 0; i < getPropCount(); i++) {
+            float x = (float) (StrictMath.round((contentScaler * getProp(i).getPosition().x + shiftX) + 0.5) - 0.5);
+            float z = (float) (StrictMath.round((contentScaler * getProp(i).getPosition().z + shiftZ) + 0.5) - 0.5);
+            float y;
+            if(heightmapBase.inBounds((int) x,  (int) z)) {
+                y = heightmapBase.getValueAt((int) x, (int) z);
+            } else {
+                y = 0; }
+            getProp(i).setPosition(new Vector3f(x, y, z));
+        }
+        for (int i = 0; i < getArmyCount(); i++) {
+            Army army = getArmy(i);
+            for (int a = 0; a < army.getGroupCount(); a++) {
+                Group group = army.getGroup(a);
+                for (int b = 0; b < group.getUnitCount(); b++) {
+                    float x = (float) (StrictMath.round((contentScaler * group.getUnit(b).getPosition().x + shiftX) + 0.5) - 0.5);
+                    float z = (float) (StrictMath.round((contentScaler * group.getUnit(b).getPosition().z + shiftZ) + 0.5) - 0.5);
+                    float y;
+                    if(heightmapBase.inBounds((int) x,  (int) z)) {
+                        y = heightmapBase.getValueAt((int) x, (int) z);
+                    } else {
+                        y = 0; }
+                    group.getUnit(b).setPosition(new Vector3f(x, y, z));
+                }
+            }
+        }
+    }
```

Comment:
A function should be made that changes the position as this is all the same operation and is a lot of duplicated code. Function could take in the list of locations or there could be a class each of these inherits from that could be used.

Reply:
Also would want to use a stream for each rather than a for loop